app/controllers/dams_resource_controller.rb

```diff
@@ -188,7 +190,22 @@ def solr_escape (str)
     pattern = /(\+|\-|\&\&|\|\||\!|\(\)|\{\}|\[|\]|\^|\"|\~|\*|\?|\:|\\)/
     str.gsub(pattern){|match|"\\"  + match} 
   end
-
+  
+  def replace_config_file (file_name)
+    copy_config_file (file_name)
+    config = ERB.new(IO.read("#{APP_ROOT}/config/#{file_name}")).result(binding)
```

Comment:
@VivianChu @mcritchlow: Looking at this function, it makes the config file loaded each time click the the push button or any other request which use it, Should we cache it?